src/nvim/syntax.c

```diff
@@ -5912,6 +5912,7 @@ static char *highlight_init_both[] =
   "default link EndOfBuffer NonText",
   "default link QuickFixLine Search",
   "default link Substitute Search",
+  "default link WhiteSpace NonText",
```

Comment:
Link to SpecialKey gives maximum compatibity to VIM.  But I think it is more reasonable to render whitesace in the same color as EOL as most other editors do by default.  Maybe this is just my personal taste. If someone prefers using SpecialKey, I could change it back.